File: backend/src/main/java/com/foodly/foodlyapi/service/ProcessService.java
package com.foodly.foodlyapi.service;

import com.foodly.foodlyapi.model.ProcessInstanceDto;
import com.foodly.foodlyapi.model.TaskHistoryDto;

import java.util.List;
import java.util.Map;

public interface ProcessService {
    
    /**
     * Aktif (devam eden) tüm süreçleri getirir
     */
    List<ProcessInstanceDto> getActiveProcesses();
    
    /**
     * Tamamlanmış tüm süreçleri getirir
     */
    List<ProcessInstanceDto> getCompletedProcesses();
    
    /**
     * Belirli bir sipariş ID'sine göre süreç detayını getirir
     */
    Map<String, Object> getProcessDetailsByOrderId(Long orderId);
    
    /**
     * Belirli bir process instance ID'sine göre süreç geçmişini getirir
     */
    List<TaskHistoryDto> getProcessHistory(String processInstanceId);
}

File: frontend/src/services/processService.js
import apiClient from './apiClient';

// Aktif süreçleri getir
export const getActiveProcesses = async () => {
  const response = await apiClient.get('/processes/active');
  return response.data;
};

// Tamamlanmış süreçleri getir
export const getCompletedProcesses = async () => {
  const response = await apiClient.get('/processes/completed');
  return response.data;
};

// Sipariş detayını getir
export const getOrderProcess = async (orderId) => {
  const response = await apiClient.get(`/processes/order/${orderId}`);
  return response.data;
};

// Süreç geçmişini getir
export const getProcessHistory = async (processInstanceId) => {
  const response = await apiClient.get(`/processes/${processInstanceId}/history`);
  return response.data;
};

export default {
  getActiveProcesses,
  getCompletedProcesses,
  getOrderProcess,
  getProcessHistory,
};
